src/Native/Unix/System.Security.Cryptography.Native/pal_bio.cpp

```diff
@@ -52,3 +52,98 @@ extern "C" int32_t CryptoNative_BioCtrlPending(BIO* bio)
     assert(result <= INT32_MAX);
     return static_cast<int32_t>(result);
 }
+
+extern "C" void CryptoNative_BioSetAppData(BIO* bio, void* data)
+{
+    BIO_set_ex_data(bio, 0, data);
+}
+
+extern "C" void CryptoNative_BioSetWriteFlag(BIO* bio)
+{
+    BIO_set_flags(bio, BIO_FLAGS_WRITE);
+}
+
+extern "C" void CryptoNative_BioSetShoudRetryReadFlag(BIO* bio)
+{
+    BIO_set_flags(bio, BIO_FLAGS_SHOULD_RETRY | BIO_FLAGS_READ);
+}
+
+typedef struct ReadWriteMethodStruct
```

Comment:
Something to me says that typedefs, structs and static data belong before any functions are declared.  I was just surprised by seeing a typedef between functions.